test/bin/pre_test_setup.sh

```diff
@@ -116,41 +72,45 @@ error_handler() {
 
 main() {
   local project_dir; project_dir="$(_get_osmt_project_dir)" || exit 135
-  local log_file; log_file="${project_dir}/api/target/osmt_spring_app.log"
 
-  # Adding docker clean up at the beginning to keep previous test state
-  # in case we need to debug and cleaning up docker could remove logs.
-  if [[ -n $(docker ps -aq --filter=name="osmt_api*") ]]; then
-    clean_docker_stack
-  fi
+  "${project_dir}/test/bin/clean_docker_stack.sh"
 
   # Run NPM install
   echo_info "Installing NPM modules..."
   install_npm_modules || exit 135
 
-  # start the API test Docker compose stack and Spring app server, detached. Send log files to 'osmt_spring_app.log'
-  echo_info "Starting OSMT Spring app for ${OSMT_STACK_NAME}. Output is suppressed, because console is detached."
-  echo_info "See 'osmt_spring_app.log' for console output. Proceeding..."
-
-  touch "$log_file"
-  "${project_dir}/osmt_cli.sh" -s  1>"$log_file" 2>"$log_file" & disown  || exit 135
-
   # curl the Spring app and retry for 2 minutes
-  curl_with_retry || exit 135
+  echo_info "Starting OSMT API Test stack..."
+  "${project_dir}/test/bin/start_osmt_app.sh" || exit 135
 
-  if [[ "${LOAD_CI_DATASET}" -eq 0  ]]; then
-    # load CI static dataset
-    "${project_dir}/osmt_cli.sh" -l || exit 135
+  echo_info "Sleeping for 5 seconds..."
+  sleep 5
 
-    # reindex ElasticSearch
-    "${project_dir}/osmt_cli.sh" -r || exit 135
+  echo_info "Loading Static CI Dataset..."
+  if [[ "${LOAD_CI_DATASET}" -eq 0  ]]; then
+    # load CI static dataset and reindex ElasticSearch
+    "${project_dir}/test/bin/load_and_reindex_ci_data.sh" || exit 135
   fi
 
   # Create postman collection
-  create_postman_collection || exit 135
+  echo_info "Creating Postman Collection for unversioned / v2 API..."
+  create_postman_collection "v2"|| exit 135
+
+  echo_info "Creating Postman Collection for unversioned / v3 API..."
+  create_postman_collection "v3"|| exit 135
 
   # Insert postman assertion tests
-  inject_tests || exit 135
+  echo_info "Injecting Postman tests for unversioned / v2 API..."
+  inject_tests \
+    "${project_dir}/test/postman/osmt-api-v2.postman_collection.json" \
+    "${project_dir}/test/postman/osmt-testing-api-v2.postman_collection.json" \
+    "v2"
+
+  echo_info "Injecting Postman tests for unversioned / v3 API..."
```

Comment:
The v3 collection is only for v3. Unversioned is handled as v2 by default at this time....